semgrep/semgrep/formatter/text.py

```diff
@@ -396,12 +406,16 @@ def format(
         cli_output_extra: out.CliOutputExtra,
         extra: Mapping[str, Any],
     ) -> str:
-        output = self._build_text_output(
-            rule_matches,
-            extra.get("color_output", False),
-            extra["per_finding_max_lines_limit"],
-            extra["per_line_max_chars_limit"],
-        )
+        reachable = []
+        unreachable = []
+        first_party = []
+        for match in rule_matches:
+            if "dependency_match_only" not in match.extra:
```

Comment:
Is this a safe check?

Either way we should add an e2e test for this new output

Reply:
How could it be unsafe? Every `rule_match` has an `extra` field. Will definitely add an e2e test though